lizard_auth_server/models.py

```diff
@@ -745,6 +749,44 @@ def as_dict(self):
 # New models
 # ##########
 
+class ProfileManager(models.Manager):
+    def get_queryset(self):
+        qs = super(ProfileManager, self).get_queryset()
+        try:
+            user = request.user
+        except RuntimeError:
+            return qs
+        if user.is_superuser:
+            return qs
+        # If the user is admin of one or more companies return all the users
+        # belonging to those companies.
+        # NOTE 1: this means an admin is able to edit users that belong to a
+        # company other than his or her own!
+        # NOTE 2: to be able to manage your own company's users you have to
+        # be explicitly admin of your company (which is in line with the
+        # implementation).
+        return qs.filter(company__in=user.profile.companies_as_admin.all())
+
+
+class CompanyManager(models.Manager):
+    def get_queryset(self):
+        qs = super(CompanyManager, self).get_queryset()
+        try:
+            user = request.user
+        except RuntimeError:
+            return qs
+        if user.is_superuser:
+            return qs
+        # Return your own company; if you're admin or guest at a company,
+        # return those companies as well.
+        company_pk = user.profile.company.pk if user.profile.company else None
+        companies_as_admin = qs.filter(
+            Q(administrators__pk__contains=user.profile.pk) |
+            Q(guests__pk__contains=user.profile.pk) |
+            Q(pk=company_pk))
+        return companies_as_admin
```

Comment:
Het is niet 100% "companies as admin", het is "companies as admin" + "companies as guest" + "company waar ik lid van ben".

Dat *is* de goede query hier, maar dan moeten er twee dingen veranderen:

- Naam van de variabele veranderen.

- Er moet nog wel filtering in de admin toegevoegd worden. Als je "edit company" rechten hebt mag je op deze manier ook alle companies editen waar je gast bent enzo. Dit kan je oplossen met object permissions (zie lizard-security). Of je zou het in de ModelAdmin kunnen oplossen. Of... ik geloof dat we tot die conclusie kwamen: "laat maar zitten want het wordt toch een custom view". Kijk maar even.